storage/encoded/encoded.go

```diff
@@ -34,26 +34,67 @@ type Codec interface {
 	Decode(w io.Writer, src []byte) error
 }
 
+// A Store wraps an existing [blob.Store] implementation so that its key spaces
+// are encoded using a [Codec].
+type Store struct {
+	codec Codec
+	real  blob.Store
+}
+
+// Keyspace implements a method of [blob.Store]. The concrete type of keyspaces
+// returned is [KV].
+func (s Store) Keyspace(name string) (blob.KV, error) {
+	kv, err := s.real.Keyspace(name)
+	if err != nil {
+		return nil, err
+	}
+	return KV{codec: s.codec, real: NewKV(kv, s.codec)}, nil
+}
+
+// Sub implements a method of [blob.Store]. The concrete type of stores
+// returned is [Store].
+func (s Store) Sub(name string) (blob.Store, error) {
+	sub, err := s.real.Sub(name)
+	if err != nil {
+		return nil, err
+	}
+	return Store{codec: s.codec, real: sub}, nil
+}
+
+// Close implements a method of [blob.StoreCloser]. It always returns nil.
+func (Store) Close(context.Context) error { return nil }
```

Comment:
Unfortunately it is, though perhaps spuriously: The test package wants a StoreCloser so we can't use that to exercise it if we don't provide that method.

I suppose I could add a NopCloser analogue, though.

Reply:
Did as noted, updated the two affected packages.

Reply:
Though I guess the Opener interface is going to want to return StoreClosers, so we may wind up wanting the adapter in the `blob` package rather than `storetest`.

Reply:
Ah, yeah. Reasonable either way.